Bottom: 9cf2ef88ac12caabad4e79c70563d1ead34e0d50
Top:    2516cf5fbc3572924f6f2139d5171f735f331563
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-01-03 11:01:11 +0800

Refresh of tuntap-xdp-transmission

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 7b8864e..2586cc3 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -2102,8 +2102,10 @@ static ssize_t tun_do_read(struct tun_struct *tun, struct tun_file *tfile,
 
 	tun_debug(KERN_INFO, tun, "tun_do_read\n");
 
-	if (!iov_iter_count(to))
+	if (!iov_iter_count(to)) {
 		tun_ptr_free(ptr);
+		return 0;
+	}
 
 	if (!ptr) {
 		/* Read frames from ring */
